Objects/typeobject.c

```diff
@@ -6630,12 +6630,18 @@ object_set_class(PyObject *self, PyObject *value, void *closure)
                 return -1;
             }
         }
+        Py_BEGIN_CRITICAL_SECTION(self);
         if (newto->tp_flags & Py_TPFLAGS_HEAPTYPE) {
             Py_INCREF(newto);
```

Comment:
Is it important to do the INCREF(newto) and DECREF(oldto) inside the critical section?

Or it it possible to restrict the critical section to the following code?

```c
oldto = Py_TYPE(self);
Py_SET_TYPE(self, newto);
```

Reply:
You're right. I can shrink it!